mautic/mautic
# Sms improvements

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [enhancement]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | 



#### Description:

This is an enhancement code that we needed to build a plugin that was adding some new tokens to SMS sending. It improves the code:
1. Reuses code that was already there and removes duplicates (the owner token handling)
2. Adds an extra SMS transport that can be used in tests. It simplifies some tests and enables testing the real sending without mocking.
3. More functional tests were added.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
4. Test that you can send SMS via Twilio and the tokens are being replaced correctly.
